controllers/openstackmachine_controller.go

```diff
@@ -281,17 +281,42 @@ func (r *OpenStackMachineReconciler) reconcileDelete(scope scope.Scope, cluster
 
 	instanceSpec := machineToInstanceSpec(openStackCluster, machine, openStackMachine, "")
 
-	if err := computeService.DeleteInstance(openStackCluster, openStackMachine, instanceStatus, instanceSpec); err != nil {
+	if err := computeService.DeleteInstance(openStackMachine, instanceStatus, instanceSpec); err != nil {
 		conditions.MarkFalse(openStackMachine, infrav1.InstanceReadyCondition, infrav1.InstanceDeleteFailedReason, clusterv1.ConditionSeverityError, "Deleting instance failed: %v", err)
 		return ctrl.Result{}, fmt.Errorf("delete instance: %w", err)
 	}
 
+	trunkEnabled, err := networkingService.IsTrunkExtSupported()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	portsStatus := openStackMachine.Status.PortsStatus
+	for _, port := range portsStatus {
+		if err := networkingService.DeleteInstanceTrunkAndPort(openStackMachine, port, trunkEnabled); err != nil {
```

Comment:
@mdbooth told me in private that this is a problem for both in an error case and on upgrade, we need to remove ports even if they are not in status.

Reply:
Matt, do we want to search ports that would have been created for this machine and try to delete them (by accepting they might not exist) ?

Reply:
I think I got my answer via https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1708/files#diff-f139d325a459bd4476f19197576c8b5b106d231872e994790c4e0842e538f264R20-R23
I'll change it to adopt ports that already exist but are not in status.